nixos/modules/services/monitoring/smartd.nix

```diff
@@ -169,12 +170,14 @@ in
         x11 = {
           enable = mkOption {
             default = config.services.xserver.enable;
+            defaultText = literalExpression "config.services.xserver.enable";
             type = types.bool;
             description = "Whenever to send X11 xmessage notifications.";
           };
 
           display = mkOption {
             default = ":${toString config.services.xserver.display}";
+            defaultText = literalExpression ''":''${toString config.services.xserver.display}"'';
```

Comment:
IMO The type conversion probably doesn't matter when you're reading the manual.

Reply:
agreed in principle, but if we're claiming that the default is "literally this expression" then the expression should probably evaluate cleanly (and be semantically equivalent to the actual default). :thinking: 

there are quite a few modules with defaultTexts that assume an implicit `with /* the config of that module */;`. might be worth adding a prefix to the manual stating that such things happens, that `toString` may be elided, etc.

Reply:
That's actually pretty fair: my brain just parsed `literalExpression` as it's own thing, ignoring the meaning in the name :P

> might be worth adding a prefix to the manual

..and, as a reminder, maybe also [search.nixos.org](https://search.nixos.org/options) but I don't think that's in nixpkgs.

Reply:
I know there's no misunderstanding anymore, but let me phrase my thoughts on the matter anyway:

`defaultText` and `literalExpression` should ideally make the text that appears in the manual literally copy-pastable in your NixOS config and attain the same result.

So yes, as it is right now is perfect.